sim/hunter/survival/explosive_shot.go

```diff
@@ -0,0 +1,74 @@
+package survival
+
+import (
+	"time"
+
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+	"github.com/wowsims/cata/sim/core/stats"
+)
+
+func (hunter *SurvivalHunter) registerExplosiveShotSpell() {
+	actionID := core.ActionID{SpellID: 53301}
+
+	hunter.Hunter.ExplosiveShot = hunter.Hunter.RegisterSpell(core.SpellConfig{
+		ActionID:    actionID,
+		SpellSchool: core.SpellSchoolFire,
+		ProcMask:    core.ProcMaskRangedSpecial,
+		Flags:       core.SpellFlagMeleeMetrics | core.SpellFlagAPL,
+
+		FocusCost: core.FocusCostOptions{
+			Cost: 50,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: time.Second,
+			},
+			IgnoreHaste: true,
+			CD: core.Cooldown{
+				Duration: time.Second * 6,
+			},
+		},
+
+		BonusCritRating: 0 +
+			core.TernaryFloat64(hunter.HasPrimeGlyph(proto.HunterPrimeGlyph_GlyphOfExplosiveShot), 6*core.CritRatingPerCritChance, 0),
+		DamageMultiplier: 1,
+		CritMultiplier: hunter.CritMultiplier(true, false, false),
+		ThreatMultiplier: 1,
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Explosive Shot - Dot",
+			},
+			NumberOfTicks: 2,
+			TickLength:    time.Second * 1,
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, isRollover bool) {
+				rap := dot.Spell.Unit.GetStat(stats.RangedAttackPower) + dot.Spell.Unit.PseudoStats.MobTypeAttackPower
+				dot.SnapshotBaseDamage = 448 + (0.273 * rap)
+				attackTable := dot.Spell.Unit.AttackTables[target.UnitIndex]
+				dot.SnapshotCritChance = dot.Spell.PhysicalCritChance(attackTable)
+				dot.SnapshotAttackerMultiplier = dot.Spell.AttackerDamageMultiplier(attackTable)
+
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeMagicHitAndSnapshotCrit)
+			},
+		},
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			result := spell.CalcAndDealOutcome(sim, target, spell.OutcomeRangedHit)
+
+			if result.Landed() {
+				spell.SpellMetrics[target.UnitIndex].Hits--
+				if spell.Dot(target).IsActive() { // Todo: Implement rollover of last tick in sim.go
+					spell.Dot(target).Rollover(sim)
+					spell.Dot(target).TickOnce(sim)
+				} else {
+					dot := spell.Dot(target)
+					dot.Apply(sim)
+					dot.TickOnce(sim)
+				}
```

Comment:
I would not try to emulate the new dot refresh here and just use the standard `dot.Apply` and leave the core to handle the rest when we update it

Reply:
Yeah, I wasn't sure if I needed to do something special here as Explosive Shot ticks once on application always, and with the new rollover, it ticks on application and then the duration is still increased.

Reply:
Fixed it yesterday to just apply and tickonce